### PR TITLE
Added MANAGED_PROVIDER to NettyClientProperties

### PR DESCRIPTION
**Pull request by msupic**

### CODE REVIEW

**Review by graemerocher — APPROVED** [review]

File: oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyClientProperties.java [hunk 1]
```diff
@@ -31,6 +31,11 @@ public final class NettyClientProperties {
      */
     public static final ClientProperty<String> SERVICE_ID = ClientProperty.create("serviceId");
 
+    /**
+     * The {@link ManagedNettyHttpProvider} to use for this client.
+     */
+    public static final ClientProperty<ManagedNettyHttpProvider> MANAGED_PROVIDER = ClientProperty.create("managedProvider");
```
Comment (yawkat) [inline oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyClientProperties.java:39]: we either make this internal, or make ManagedNettyHttpProvider public

File: oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyHttpClientBuilder.java [hunk 2]
```diff
@@ -74,6 +74,8 @@ public <T> HttpClientBuilder property(ClientProperty<T> key, T value) {
             key == StandardClientProperties.HOSTNAME_VERIFIER ||
             key == StandardClientProperties.SSL_CONTEXT) {
             throw new IllegalArgumentException("The OCI SDK netty client does not support changing the this setting (" + key + ") directly. Please go through the Micronaut HTTP client configuration.");
+        } else if (key == NettyClientProperties.MANAGED_PROVIDER) {
+            managedProvider = (ManagedNettyHttpProvider) value;
```
Comment (yawkat) [inline oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyHttpClientBuilder.java:78]: please add a check that managedProvider is null

**Review by yawkat — APPROVED** [review]

File: oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyHttpClientBuilder.java (hunk 2 above)
Comment (msupic) [reply oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyHttpClientBuilder.java:78]: Done

File: oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyClientProperties.java (hunk 1 above)
Comment (msupic) [reply oraclecloud-httpclient-netty/src/main/java/io/micronaut/oraclecloud/httpclient/netty/NettyClientProperties.java:39]: Done

**Pull request merged**